staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/metrics.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"time"
+
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+const (
+	namespace = "apiextensions_apiserver"
+	subsystem = "validation"
+)
+
+var Metrics = newValidationMetrics()
+
+type ValidationMetrics struct {
+	ratchetingTime *metrics.Histogram
+}
+
+func newValidationMetrics() *ValidationMetrics {
+	m := &ValidationMetrics{
+		ratchetingTime: metrics.NewHistogram(&metrics.HistogramOpts{
+			Namespace:      namespace,
+			Subsystem:      subsystem,
+			Name:           "ratcheting_seconds",
+			Help:           "Time for comparison of old to new for the purposes of CRDValidationRatcheting during an UPDATE in seconds.",
+			StabilityLevel: metrics.ALPHA,
+			// Start 0.01ms with the last bucket being [~2.5s, +Inf)
```

Comment:
is 2.5s enough?

Reply:
IMO, 2.5s is basically infinite time yeah. I have very little patience.

Reply:
I think so. For the comparison to be on the order of seconds is already very concerning. IMO higher buckets wouldn't give much more information than 2.5s